prive/datasets/dataset.py

```diff
@@ -63,76 +65,210 @@ def replace(self, record_in, record_out):
+        # dtypes = {cd['name']: _get_dtype(cd) for cd in self.description['columns']}
+        # columns = self.description['columns']
 
-        dataset = pd.read_csv(f'{input_path}.csv')
+        dataset = pd.read_csv(f'{filepath}.csv')
 
-        return dataset, description
+        return cls(dataset, description)
 
-    def write(self, output_path):
+    def write(self, filepath):
         """
         Write dataset and description to file
+
+        Parameters
+        ----------
+        filepath (str): Path where the csv and json file are saved
         """
-        pass
+
+        with open(f'{filepath}.csv', 'w') as fp:
+            json.dump(dict, fp)
+
+        self.dataset.to_csv(filepath)
 
     def sample(self, n_samples):
         """
-        Sample from dataset a set of records.
-        """
-        pass
+        Sample from a TabularDataset object a set of records.
+
+        Parameters
+        ----------
+        n_samples (int): Number of records to sample
+
+        Returns
+        -------
+
+        A TabularDataset object with a sample of the records of the original object.
+
 
-    def get_record(self, record_ids):
         """
-        Select and return a record(s).
+        return TabularDataset(dataset=self.dataset.sample(n_samples), description=self.description)
+
+    def get_records(self, record_ids):
         """
-        pass
+        Get a record from the TabularDataset object
+
+        Parameters
+        ----------
+        record_ids (List[int]): List of indexes of records to retrieve
+
+        Returns
+        -------
+
+        A TabularDataset object with the record(s).
 
-    def drop_record(self, record_ids):
         """
-        Drop a record(s) and return modified dataset.
+
+        # TODO: what if the index is supposed to be a column? an identifier?
+        return TabularDataset(self.dataset.iloc[record_ids], self.description)
+
+    def drop_records(self, record_ids):
         """
-        pass
+        Get a record from the TabularDataset object
+
+        Parameters
+        ----------
+        record_ids (List[int]): List of indexes of records to drop
 
+        Returns
+        -------
 
-    def add_record(self, record):
+        A TabularDataset object without the record(s).
+
+        """
+        # TODO: what if the index is supposed to be a column? an identifier?
+        return TabularDataset(self.dataset.drop(record_ids), self.description)
+
+    def add_records(self, records):
         """
         Add record(s) to dataset and return modified dataset.
+
+        Parameters
+        ----------
+        records (TabularDataset): A TabularDataset object with the record(s) to add.
+
+        Returns
+        -------
+
+        A new TabularDataset object with the record(s).
         """
-        pass
 
-    def replace(self, record_in, record_out):
+        # this does the same as the __add__
+        return self.__add__(records)
+
+    def replace(self, records_in, records_out):
```

Comment:
It would also be good if this method allowed for a random record to be removed, which could happen when `records_out=None`, and that should be the default.

Reply:
inplace left for next PR, replace and drop changed to add random record to be removed.